README.md

```diff
@@ -1,4 +1,208 @@
+
+The **structure** is written in terms of modular UI blocks (see below) assembled in a declarative UI file.
+This declarative file is a JSON file.
+However, we include a preprocessor tool called Autobahn so that you can write your app description in a more expressive, human-friendly YAML format.
+See `data/preset/*.yaml` for examples.
+
+The **theme** is a GTK CSS file.
+We suggest writing themes using SCSS and then compiling them down to regular CSS.
+We provide SCSS modules that your SCSS code can import; see `data/css/`.
+
+The **content** is in the form of a shard file; see eos-shard for more information.
+We are working on tools that allow you to build shard files easily on your own machine.
+For now, we include a testing tool called Moltres that will run your app with sample content.
+To try it out, run it on one of the app description YAML files in `data/preset/`: e.g. `moltres data/preset/buffet.yaml`.
+
+Finding your way around
+-----------------------
+Here's an overview of what's in each directory:
+
+- `data/`
+  - `css/` - themes and parts of themes for UI modules
+  - `images/` - miscellaneous UI assets
+  - `moltres/` - sample assets for the Moltres app runner tool
+  - `preset/` - pre-built YAML UI description files
+  - `templates/` - code for rendering HTML content; more or less deprecated
+  - `widgets/` - GtkBuilder XML files for UI modules in `js/app/modules/`
+- `docs/`
+  - Various Markdown developer docs
+  - `reference/js/` - API reference; not useful right now, being rewritten.
+- `js/` - Javascript library code
+  - `app/` - the main UI library for creating apps
+    - Various app library classes
+    - `interfaces/` - interfaces which some UI modules implement
+    - `libs/` - external JS code such as Mustache
+    - `modules/` - the modular UI blocks (see below)
+    - `widgets/` - some GTK widgets for UI code that are not modular UI blocks
+  - `search/` - library for accessing databases of offline content
+  - `tools/` - various testing and development tools
+- `lib/` - C library code
+  - `eosknowledgeprivate/` - introspectable C wrappers for `js/app/`
+  - `eosknowledgesearchprivate/` - introspectable C wrappers for `js/search/`
+  - `web-extensions/` - WebKit web extensions for use when viewing documents
+- `m4/` - Autoconf code
+- `po/` - internationalization (see below)
+- `tests/` - unit and integration tests
+  - Various utility code for tests
+  - `autobahn/` - integration tests for the Autobahn YAML processor
+  - `js/` - unit tests for the `js/app` and `js/search` libraries
+  - `test-content/` - sample content for some tests
+
+Modular UI blocks
+-----------------
+UI modules are GObjects, with a few extra features and a few restrictions.
+They are meant to be assembled in a declarative file, without writing having to write any code such as signal handlers.
+
+First of all there are no signals, because there is no code.
+(A UI module's behavior is baked into its code.)
```

Comment:
This incidental inside the parentheses is unnecessary.

"First of all, UI modules do not emit signals, because their behavior is part of their implementation, and UI descriptions are purely declarative."